### PR TITLE
Assignment5 with @anthonehi

### PR DESCRIPTION
**Pull request by roylalog000**
code pairing @anthonehi

### CODE REVIEW

**Comment by jmplett** [comment]
I will review

File: SecretSanta/src/SecretSanta.Api/Controllers/UsersController.cs [hunk 1]
```diff
@@ -43,33 +53,41 @@ public ActionResult Delete(int id)
...
                 return BadRequest();
             }
-            return Repository.Create(user);
+            return Repository.Create(myUser);
         }
 
+
         [HttpPut("{id}")]
-        public ActionResult Put(int id, [FromBody] User? user)
+        public ActionResult Put(int id, [FromBody]UpdatedUser? updatedEvent)
```
Comment (jmplett) [inline SecretSanta/src/SecretSanta.Api/Controllers/UsersController.cs:70]: Missing status codes

File: SecretSanta/src/SecretSanta.Web/Controllers/UsersController.cs [hunk 2]
```diff
@@ -1,36 +1,62 @@
...
             return View(viewModel);
         }
 
+
+
+
         public IActionResult Edit(int id)
         {
-            return View(MockData.Users[id]);
+            return View(MockData.Users[id - 1]);
```
Comment (jmplett) [inline SecretSanta/src/SecretSanta.Web/Controllers/UsersController.cs:59]: Needs to be implemented with the DTO

File: SecretSanta/test/SecretSanta.Api.Tests/Controllers/UserControllerTests.cs [hunk 3]
```diff
@@ -15,6 +15,6 @@ namespace SecretSanta.Api.Tests.Controllers
     [TestClass]
     public class UserControllerTests
     {
-        
+
```
Comment (jmplett) [inline SecretSanta/test/SecretSanta.Api.Tests/Controllers/UserControllerTests.cs:18]: Tests are not done

**Comment by jmplett** [comment]
- Update the API project with the following
  - Add a new Dto namespace ✔
  - The `UserController` should only accept or return classes from the Dto namespace. Create any needed classes there and update the controller to use them. You will need to convert between the DTO classes and the relevant entities as needed. ❌
  - Add OpenAPI (swagger) endpoint ✔
  - Use the OpenAPI to generate a C# client with relevant interfaces ✔
  - Decorate all relevant API endpoints in the `UserController` with the appropriate [ProducesResponseType](https://docs.microsoft.com/en-us/aspnet/core/web-api/action-return-types?view=aspnetcore-5.0#iactionresult-type) ❌
- Update the Web project with the following
  - Register the generated `UserClients` class using its interface with the Dependency Injection (DI) service. ❌
  - Update the `UserController` to take in the generated client as a dependency. ❌
  - Convert the `UserController` from using the `MockData` to using the client to get data from the API service. ❌
  - Update the controller to convert back and forth between the ViewModel classes and the API DTO classes as needed. ❌
- Update the SecretSanta.Api.Tests with the following:
  - The controller tests should be [interation tests](https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-5.0) not unit tests. ❌
  - All endpoints should be tested. ❌
  - The `UserController` should use a test double for the `IUserRepository` to avoid needing to retest business and data logic. ❌
- Create a new SecretSanta.Web.Tests projecs with the following:
  - It should be inside of the test directory. ✔
  - It should have nullable reference types enabled. ✔
  - It should have functional tests for all endpoints in the `UserController` that access the API. ❌
  - It should use a test double for the API service. This test double should be used to validate the controller's interaction with the API without actually calling the API. ❌

Good start just have some loose ends to fix and them complete the rest of the assignment.

**Comment by Keboo** [comment]
@jmplett though perfectly fine to review, when possible see if there is a PR that was not done as a [pair](https://github.com/IntelliTect-Samples/EWU-CSCD379-2021-Spring/blob/main/Docs/Homework.md#pairing) when looking to review :).

**Pull request closed without merging**